pkg/schedule/schedulers/balance_range.go

```diff
@@ -86,6 +180,48 @@ type balanceRangeSchedulerJob struct {
 	Status  JobStatus       `json:"status"`
 }
 
+func (conf *balanceRangeSchedulerConfig) deleteJob(jobID uint64) error {
+	conf.Lock()
+	defer conf.Unlock()
+	for _, job := range conf.jobs {
+		if job.JobID == jobID {
+			status := job.Status
+			if job.Status != pending && job.Status != running {
+				return errs.ErrScheduleConfigNotExist.FastGenByArgs(jobID)
+			}
+			job.Status = cancelled
+			now := time.Now()
+			job.Start = &now
+			job.Finish = &now
+			if err := conf.save(); err != nil {
+				job.Status = status
+				job.Start = nil
+				job.Finish = nil
+				return err
+			}
```

Comment:
Do we need to recover `start` and `finish` to previous value when meet error?

Reply:
Good catch, the start should be recovered. fixed